sounddevice.py

```diff
@@ -2349,6 +2376,125 @@ def __init__(self, channel_selectors):
             channelSelectors=self._selectors))
 
 
+class CoreAudioSettings(object):
+
+    def __init__(self, channel_map=None, change_device_parameters=False,
+                 fail_if_conversion_required=False,
+                 conversion_quality='Max'):
+        """Mac Core Audio-specific input/output settings.
+
+        Objects of this class can be used as *extra_settings* argument
+        to `Stream()` (and variants) or as `default.extra_settings`.
+
+        Parameters
+        ----------
+        channel_map : array of int, optional
+            Support for opening only specific channels of a MacCore
+            device.  *channel_map* is a list of integers specifying the
+            (zero-based) channel numbers to use.  Note that
+            *channel_map* is treated differently between input and
+            output channels.
+
+            For output devices, *channel_map* must have the same length
+            as the number of channels of the device.  Initialize unused
+            channels in this array with -1, and a 0-based index for any
+            desired channels.  For an example 6-channel output device
+            where an application wants to only write to the third and
+            fourth channels, this array would be [-1, -1, 0, 1, -1, -1].
+
+            For input devices, *channel_map* is a list of integers
+            specifying the (zero-based) channel numbers to use.  For an
+            example 6-channel input device where the application wants
+            to read from the third and fourth channels, this array would
+            be [2, 3].
+
+            The example below demonstrates setting up for a 94-channel
+            device.  For additional information, see
+            https://app.assembla.com/spaces/portaudio/git/source/master/src/hostapi/coreaudio/notes.txt
+        change_device_parameters : bool, optional
+            If ``True``, allows PortAudio to change things like the
+            device's frame size, which allows for much lower latency,
+            but might disrupt the device if other programs are using it,
+            even when you are just Querying the device.  ``False`` is
+            the default.
+        fail_if_conversion_required : bool, optional
+            In combination with the above flag, ``True`` causes the
+            stream opening to fail, unless the exact sample rates are
+            supported by the device.
+        conversion_quality : string, optional
```

Comment:
Actually, I'm not a fan of long names, but in this case I think it is better to follow the sub-optimal PortAudio naming instead of coming up with our own names which may have problems on their own. This way we can at least blame it on PortAudio.

I am very anal about line lengths, but that doesn't mean I would never accept exceptions.
In this case I would make an exception.